genai-function-calling/vercel-ai/package.json

```diff
@@ -7,13 +7,14 @@
     "node": ">=22"
   },
   "scripts": {
-    "start": "node --env-file .env -r @elastic/opentelemetry-node index.js"
+    "start": "node --env-file .env -r @elastic/opentelemetry-node -r ./telemetry.js index.js"
```

Comment:
Not sure it matters now that you've gone back to CommonJS, but EDOT Node.js now recommends using
`node --import @elastic/opentelemetry-node ...` rather than `node -r @elastic/opentelemetry-node`. Only the former will enable instrumentation of ES modules.  Using `--import ...` is fine and safe to use even if you are using CommonJS for your code.

Anyway, I doubt this is related to the issues you are currently seeing.

Reply:
fyi I rewrote this comment with a lengthy overview which might help https://github.com/elastic/observability-examples/pull/64#issuecomment-2822722800

Reply:
thanks for the tip. I switched to --import and verify base case (`npm run start`) makes a trace connected (not using MCP), and as you guessed it doesn't impact `npm run start -- --mcp` which is disconnected still.